src/java.base/share/classes/java/util/stream/Gatherer.java

```diff
@@ -257,6 +257,10 @@ default BiConsumer<A, Downstream<? super R>> finisher() {
      * Returns a composed Gatherer which connects the output of this Gatherer
      * to the input of that Gatherer.
      *
+     * @implSpec The implementation in this interface returns a new Gatherer
+     *           which is semantically equivalent to the combination of
+     *           {@code this} and {@code that} gatherer.
+     *
```

Comment:
@AlanBateman @PaulSandoz `Gatherer#andThen` was missing an @implSpec, so this commit addresses that.